packages/select/src/Select/Select.tsx

```diff
@@ -39,6 +39,14 @@ export interface SelectProps
     ref: React.RefObject<HTMLButtonElement>
   ) => React.ReactNode
   children?: (optionsProps: SelectOptionsProps) => React.ReactNode
+  /**
+   * Updates the styling of the validation FieldMessage.
+   */
+  status?: "error" | "caution"
+  /**
+   * A descriptive message for the 'status' states.
+   */
+  validationMessage?: React.ReactNode | undefined
```

Comment:
Should we make the validation message required if there's a status? There doesn't seem to be a default 🤔

Reply:
`type ValidationProps =
  | {
      status: "error" | "caution"
      validationMessage: React.ReactNode | undefined
    }
  | {
      status?: never
      validationMessage?: never
    }`

Reply:
```
type ValidationProps =
  | {
      status: "error" | "caution"
      validationMessage: React.ReactNode | undefined
    }
  | {
      status?: never
      validationMessage?: never
    }
    
    ```

Reply:
I could do this and then `export const Select: React.FC<SelectProps & ValidationProps>`

Reply:
thoughts @gyfchong ? Means I can only have status + validation message or niether

Reply:
Hmm I just had a look at the others, doesn't seem like we add any restriction on this, so nevermind in this case. I'll pose the question for a more universal change 👍